### ISSUE TITLE
TSGateway Server disconnect

### ISSUE THREAD

**Issue by MartinHaimberger** [FreeRDP/FreeRDP#1194]
If the TSGateway server sends a disconnect (either admin disconnect or server shutdown (see TSGW spec. below)) the client has to shut down the connection.

Now the freerdp client IGNORES the final response packet and this leeds to a hungup of the client.

2.2.3.4.3 TSG Server to TSG Client Packet Format for Final Response
This is the final response from the TSG server to the TSG client. To indicate connection disconnect, TSG server MUST set the PFC_LAST_FRAG bit in pfc_flags of the header of the RPC response PDU as described in TsProxySetupReceivePipe (section 3.1.4.2.2). For a description of RPC response PDU, pfc_flags, PFC_LAST_FRAG, and stub data, refer to sections 12.6.2 and 12.6.4.10 in [C706]. PDU body contains the return value as shown in the following packet diagram.

3.1.4.3.4 Server Initiated Shutdown
The server initiates shutdown by sending the final response packet to TsProxySetupReceivePipe call with the PFC_LAST_FRAG bit set in the pfc_flags field. The server closes the channel after sending this response. The client SHOULD not call TsProxyCloseChannel after receiving this final response. The client SHOULD call the TsProxyCloseChannel method if the client initiates the shutdown, but not if the server initiates shutdown.
Prerequisites: The connection MUST be in Pipe Created state.
Sequential Processing Rules:
1. The TSG server MUST send the final response packet to TsProxySetupReceivePipe call with the PFC_LAST_FRAG bit set in the pfc_flags field.
2. The TSG server MUST close the channel.
3. The connection MUST be transitioned to Tunnel Close Pending state.

Nice regards,
Martin Haimberger

**Comment by akallabeth** [comment]
Gateway support has been hugely improved with `2.0.0-rc4` and #5022 
Closing as no longer applicable.
Create a new issue if still a problem.
